docs/executor-custom.md

```diff
@@ -101,11 +162,209 @@ echo "http://google.pl" | kubectl testkube tests create --name example-google-te
+  error("No valid data directory detected");
+  process.exit(1);
+}
+
+const testContentPath = path.join(process.env.RUNNER_DATADIR, "test-content");
+uri = fs.readFileSync(testContentPath, { encoding: "utf8", flag: "r"});
+
+https.get(uri, (res) => {
+    if (res.statusCode == 200) {
+      successResult("Got valid status code: 200 OK");
+    } else {
+      errorResult("Got invalid status code");
+    }
+  })
+  .on("error", (err) => { error("Error: " + err.message); });
+
+
+function errorResult(message) {
+  console.log(JSON.stringify({ "type": "result", "result": { "status": "error", "errorMessage": message, }}));
+}
+
+function successResult(output) {
+  console.log(JSON.stringify({ "type": "result", "result": { "status": "success", "output": output, }}));
+}
+
+// error will return error info not related to test itself (some issues with executor)
+function error(message) {
+  console.log(JSON.stringify({ "type": "error", "content": message, })); 
+}
+  
+```
+
+Code is ready and working - we're assuming defaults so `RUNNER_DATADIR` will be `/data` and our file will be in `/data/test-content` file.
+
+As we can see we're pushing JSON output to stdin with console.log function (it's based on our [OpenAPI spec - ExecutorOutput](https://kubeshop.github.io/testkube/openapi/))
+
+Two basic output types are handled here:
+- in case of executor failures (non-test related) we should return `error`, 
+- in case of test result we should return `result` with test status (success, error)
+
+
+Now when executor code is ready we need additional steps:
+- Docker image (create image and push)
+- Kubernetes Executor Custom Resource definition
+- Create test
+
+Let's start with Docker: 
+We'll simplify and use latest tag here - but you should use versioning as good practice.
+As for now testkube runs directly command and is passing execution information as argument 
+We need to add runner binary (but we have plans to remove need of this step)
+
+```sh
+#!/usr/bin/env sh
+node app.js "$@"
+```
+
+And add it into our Dockerfile
+
+```Dockerfile
+FROM node:17
+
+# Create app directory
+WORKDIR /usr/src/app
+
+# Bundle app source
+COPY runner /bin/runner
+RUN chmod +x /bin/runner
+COPY app.js app.js
+
+EXPOSE 8080
+CMD [ "/bin/runner" ]
+```
+
+Now let's build and push our docker container (change user/repo to your Docker Hub username): 
+```sh
+docker build --platform=linux/amd64 -t USER/testkube-executor-example-nodejs:latest -f Dockerfile .
+docker push USER/testkube-executor-example-nodejs:latest
+```
+
+After we have our image in place where Kubnernetes can load it we need to define our executor: 
+
+```yaml
+apiVersion: executor.testkube.io/v1
+kind: Executor
+metadata:
+  name: example-nodejs-executor
+  namespace: testkube
+spec:
+  executor_type: job
+  features: []
+  image: kubeshop/testkube-executor-example-nodejs:latest
+  types:
+    - example/test
+```
+
+Save it somewhere e.g. `example-executor.yaml` and apply into Kubernetes cluster: 
```

Comment:
5. Save the file to a file name, e.g., `example-executor.yaml`, and apply it into the Kubernetes cluster: